### PR TITLE
feat: update PATCH /connectors/:id

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

update PATCH /connectors/:id
we can now also update `metadata` via this API (before this change, we can only update `config`; and `metadata` is a new db field).

## Testing

UTs and ITs.

### CODE REVIEW

**Comment by darcyYe** [comment]
Need to update @logto/connector-kit dependency (see [PR of @logto/toolkit](https://github.com/logto-io/toolkit/pull/37)).

File: packages/core/src/routes/connector.ts [hunk 1]
```diff
@@ -146,8 +148,22 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
         validateConfig(body.config);
       }
 
+      if (metadata.isStandard === true) {
+        const databaseMetadataGuard = configurableConnectorMetadataGuard.required();
+        const result = databaseMetadataGuard.safeParse(body.metadata);
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:153]: To align with `ConnectorResponse`, better remove "metadata" and spread its elements.

**Review by wangsijie — COMMENTED** [review]
The PR description can be more specific on what functions have been added or modified for the "update".

File: packages/core/src/routes/connector.ts (hunk 1 above)
Comment (darcyYe) [reply packages/core/src/routes/connector.ts]: Current update-related APIs response body is not aligned with ConnectorResponse type, maybe fix this later.

File: packages/core/src/routes/connector.ts [hunk 2]
```diff
@@ -132,18 +132,35 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
+        validateConfig,
+      } = await getLogtoConnectorById(id);
+
+      if (config) {
+        validateConfig(config);
+      }
 
-      if (body.config) {
-        validateConfig(body.config);
+      if (isStandard === true) {
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:156]: for a PATCH endpoint, the user don't have to set `metadata`, consider add an if statement

File: packages/core/src/routes/connector.ts [hunk 3]
```diff
@@ -132,18 +132,35 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
+        assertThat(
+          databaseMetadata?.target &&
+            databaseMetadata.name &&
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:159]: As the db's metadata will be merged with virtual connector's metadata, maybe only `target` is required?

File: packages/core/src/routes/connector.ts [hunk 4]
```diff
@@ -132,20 +132,24 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
 
       const { metadata, type, validateConfig } = await getLogtoConnectorById(id);
 
-      if (body.config) {
-        validateConfig(body.config);
+      if (config) {
+        validateConfig(config);
       }
 
+      // TODO[LOG-4696]: revisit databaseMetadata check when implementing AC
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:151]: nit: use `// TODO @somebody` or `// TODO LOG-4696`

**Review by wangsijie — APPROVED** [review]

**Pull request merged**